core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java

```diff
@@ -0,0 +1,245 @@
+    final int FILES_PINNED_TEST_VALUE = 100;
+    String filesPinnedProperty =
+        MetricsSystem.getMasterMetricName(FileSystemMaster.Metrics.FILES_PINNED);
+    Gauge<Integer> filesPinnedGauge = new Gauge<Integer>() {
+      @Override
+      public Integer getValue() {
+        return FILES_PINNED_TEST_VALUE;
+      }
+    };
+    MetricSet mockMetricsSet = mock(MetricSet.class);
+    Map<String, Metric> map = new HashMap<>();
+    map.put(filesPinnedProperty, filesPinnedGauge);
+
+    when(mockMetricsSet.getMetrics()).thenReturn(map);
+    MetricsSystem.METRIC_REGISTRY.registerAll(mockMetricsSet);
+
+    Response response = mHandler.getMetrics();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertTrue("Entry must be a SortedMap!", (response.getEntity() instanceof SortedMap));
+    SortedMap<String, Long> entry = (SortedMap<String, Long>) response.getEntity();
+    assertFalse("Properties Map must be not empty!", (entry.isEmpty()));
+    assertTrue("Map must contain key " + filesPinnedProperty + "!",
+        entry.containsKey(filesPinnedProperty));
+    assertEquals(FILES_PINNED_TEST_VALUE, entry.get(filesPinnedProperty).longValue());
+  }
+
+  @Test
+  public void getStartTimeMs() {
+    Response response = mHandler.getStartTimeMs();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertEquals("Entry must be a Long!", Long.class, response.getEntity().getClass());
+    Long entry = (Long) response.getEntity();
+    assertEquals(0L, entry.longValue());
+
+    when(mMaster.getStartTimeMs()).thenReturn(100L);
+    response = mHandler.getStartTimeMs();
+    entry = (Long) response.getEntity();
+    assertEquals(100L, entry.longValue());
+  }
+
+  @Test
+  public void getUptimeMs() {
+    Response response = mHandler.getUptimeMs();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertEquals("Entry must be a Long!", Long.class, response.getEntity().getClass());
+    Long entry = (Long) response.getEntity();
+    assertEquals(0L, entry.longValue());
+
+    when(mMaster.getUptimeMs()).thenReturn(100L);
+    response = mHandler.getUptimeMs();
+    entry = (Long) response.getEntity();
+    assertEquals(100L, entry.longValue());
+  }
+
+  @Test
+  public void getVersion() {
+    Response response = mHandler.getVersion();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertEquals("Entry must be a String!", String.class, response.getEntity().getClass());
+    String entry = (String) response.getEntity();
+    assertEquals("\"" + RuntimeConstants.VERSION + "\"", entry);
+  }
+
+  @Test
+  public void getCapacityBytes() {
+    Response response = mHandler.getCapacityBytes();
```

Comment:
We often use `PowerMockito` to mock final classes in cases like this. To use it you would annotate the start of the test with 
```java
@RunWith(PowerMockRunner.class)
@PrepareForTest(BlockMaster.class)
```
and then you could use `PowerMockito.mock(BlockMaster.class)` exactly how you would use `Mockito.mock(BlockMaster.class)` if it wasn't final.